aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -90,6 +90,10 @@ def status(all_profiles):
     else:
         profiles = [manager.get_profile()]
 
+    if not profiles:
+        echo.echo_error('No profiles found')
```

Comment:
You can use `echo.echo_critical` instead, which relieves you of the need to manually `sys.exit`

Reply:
This fix won't solve the issue I am afraid (which shows why we should really be adding a test for this). The problem is that `manager.get_profile()` can return `None` and so `profiles == [None]` and `if [None]` evaluates to `True`. So the conditional `if no profiles` won't be hit.

Reply:
Hehe, thanks for catching!

I didn't know myself where it would be easiest to add a test for this; do you know?

Reply:
Please add the following test at the end of the `tests/cmdline/commands/test_daemon.py` file:
```python
@pytest.mark.usefixtures('empty_config')
def test_daemon_status_no_profile(run_cli_command):
    """Test ``verdi daemon status`` when no profiles are defined."""
    result = run_cli_command(cmd_daemon.status, raises=True)
    assert 'No profiles defined.' in result.output

```
This should currently fail. But if the command properly calls `echo.echo_critical(No profiles defined.)` if no profiles are defined, the test will pass